crucible/src/Lang/Crucible/CFG/Expr.hs

```diff
@@ -983,16 +977,16 @@ data App (ext :: Type) (f :: CrucibleType -> Type) (tp :: CrucibleType) where
   -- If no such position exists, return a negative value.
   StringIndexOf :: !(f (StringType si))
                 -> !(f (StringType si))
-                -> !(f NatType)
+                -> !(f IntegerType)
                 -> App ext f IntegerType
 
   -- @stringSubstring s off len@ extracts the substring of @s@ starting at index @off@ and
   -- having length @len@.  The result of this operation is undefined if @off@ and @len@
   -- do not specify a valid substring of @s@; in particular, we must have @off+len <= length(s)@.
   StringSubstring :: !(StringInfoRepr si)
                   -> !(f (StringType si))
-                  -> !(f NatType)
-                  -> !(f NatType)
+                  -> !(f IntegerType)
+                  -> !(f IntegerType)
```

Comment:
Same evaluation concern here for these two arguments as with the `StringIndexOf` argument above.